testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestCreateDropSchema.java

```diff
@@ -39,18 +42,159 @@
     @Test
     public void testCreateDropSchema()
     {
-        onHive().executeQuery("DROP DATABASE IF EXISTS test_drop_schema CASCADE");
+        String schemaName = "test_drop_schema";
+        String schemaDir = warehouseDirectory + "/test_drop_schema.db";
 
-        onTrino().executeQuery("CREATE SCHEMA test_drop_schema");
-        assertTrue(hdfsClient.exist(warehouseDirectory + "/test_drop_schema.db"));
+        ensureSchemaDoesNotExist(schemaName);
+
+        assertQuerySucceeds("CREATE SCHEMA test_drop_schema");
+        assertThat(hdfsClient.exist(schemaDir))
+                .as("Check if expected schema directory exists after creating schema")
+                .isTrue();
 
         onTrino().executeQuery("CREATE TABLE test_drop_schema.test_drop (col1 int)");
+
         assertQueryFailure(() -> query("DROP SCHEMA test_drop_schema"))
                 .hasMessageContaining("line 1:1: Cannot drop non-empty schema 'test_drop_schema'");
 
         onTrino().executeQuery("DROP TABLE test_drop_schema.test_drop");
 
-        onTrino().executeQuery("DROP SCHEMA test_drop_schema");
-        assertFalse(hdfsClient.exist(warehouseDirectory + "/test_drop_schema.db"));
+        assertQuerySucceeds("DROP SCHEMA test_drop_schema");
+        assertThat(hdfsClient.exist(schemaDir))
+                .as("Check if schema directory exists after dropping schema")
+                .isFalse();
+    }
+
+    @Test
+    public void testDropSchemaWithEmptyLocation()
+    {
+        String schemaName = schemaName("schema_with_empty_location");
+        String schemaDir = warehouseDirectory + "/schema-with-empty-location/";
+
+        createSchema(schemaName, schemaDir);
+        assertFileExists(schemaDir, true, "schema directory exists after create schema");
+        assertQuerySucceeds("DROP SCHEMA " + schemaName);
+        assertFileExists(schemaDir, false, "schema directory exists after drop schema");
+    }
+
+    @Test
+    public void testDropSchemaFilesWithoutLocation()
+    {
+        String schemaName = schemaName("schema_without_location");
+        String schemaDir = format("%s/%s.db/", warehouseDirectory, schemaName);
+
+        createSchema(schemaName);
```

Comment:
inlining `createSchema` would make it crystal-clear that `CREATE SCHEMA` has no `WITH (location ..)`